src/components/CountryPicker/index.js

```diff
@@ -0,0 +1,92 @@
+import React, {useState, useEffect} from 'react';
+import {View} from 'react-native';
+import PropTypes from 'prop-types';
+import styles from '../../styles/styles';
+import MenuItemWithTopDescription from '../MenuItemWithTopDescription';
+import * as PersonalDetails from '../../libs/actions/PersonalDetails';
+import useLocalize from '../../hooks/useLocalize';
+import CountrySelectorModal from './CountrySelectorModal';
+import FormHelpMessage from '../FormHelpMessage';
+
+const propTypes = {
+    /** The ISO code of the country */
+    countryISO: PropTypes.string,
+
+    /** Form Error description */
+    errorText: PropTypes.string,
+
+    /** Country to display */
+    // eslint-disable-next-line react/require-default-props
+    value: PropTypes.string,
+
+    /** Callback to call when the input changes */
+    onInputChange: PropTypes.func,
+
+    /** Callback to call when the Country is updated */
+    onCountryUpdated: PropTypes.func,
+};
+
+const defaultProps = {
+    countryISO: '',
+    errorText: '',
+    onInputChange: () => {},
+    onCountryUpdated: () => {},
+};
+
+const CountryPicker = React.forwardRef(({value, countryISO, errorText, onCountryUpdated, onInputChange}, ref) => {
```

Comment:
This component uses two values (value, countryISO) and two callbacks (onCountryUpdated, onInputChange). This can be simplified. Keep only `value` and `onInputChange`.

Those are coming from `AddressPage` Form.. `<Form />` will call `onValueChange` callback for each input, we can make use of that for both CountryPicker and AddressSearch.

**CountryPicker** (to cover the case where the country is changed from the picker)

```js
onValueChange={setCurrentCountry}
```

**AddressSearch** (to cover the case where the country is changed from the google autocomplete)

```js
onValueChange={(value, key) => {
    if (key !== 'country') {
        return;
    }
    setCurrentCountry(value);
}}
```

___

We also need to make a little change to Form to pass the key:
https://github.com/Expensify/App/blob/0b00e1cc82a4f6023aff6b89796a91fba6be91a2/src/components/Form.js#L318-L320